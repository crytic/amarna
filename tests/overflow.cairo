# Adds two integers. 
# Reverts if the sum overflows.
func add{
        syscall_ptr: felt*,
        pedersen_ptr: HashBuiltin*,
        range_check_ptr
    } (a: Uint256, b: Uint256) -> (c: Uint256):
    uint256_check(a)
    uint256_check(b)
    let (c: Uint256, is_overflow) = uint256_add(a, b)
    with_attr error_message("SafeUint256: addition overflow"):
        assert is_overflow = FALSE
    end
    return (c)
end

# Adds two integers. 
# Reverts if the sum overflows.
func add{
        syscall_ptr: felt*,
        pedersen_ptr: HashBuiltin*,
        range_check_ptr
    } (a: Uint256, b: Uint256) -> (c: Uint256):
    uint256_check(a)
    uint256_check(b)
    let (c: Uint256, _) = uint256_add(a, b)
    return (c)
end