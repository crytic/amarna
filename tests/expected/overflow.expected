[must-check-overflow] in overflow.cairo:10:22
[must-check-overflow] in overflow.cairo:26:22
[unused-function] in overflow.cairo:19:1
[unused-function] in overflow.cairo:3:1